stdlib/src/builtin/bool.mojo

```diff
@@ -322,6 +327,15 @@ struct Bool(
         """
         return __mlir_op.`index.casts`[_type = __mlir_type.index](self.value)
 
+    @always_inline("nodebug")
+    fn __index__(self) -> Int:
+        """Convert this Bool to an integer for indexing purposes
+
+        Returns:
+            Bool as Int
```

Comment:
Please provide a more details description here, e.g. "`1` if `True` and `0` otherwise."